src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.GeneratedFileReplacingCompilationTracker.cs

```diff
@@ -57,6 +57,13 @@ public ICompilationTracker Fork(ProjectState newProject, CompilationAndGenerator
                 throw new NotImplementedException();
             }
 
+            public ICompilationTracker FreezePartialState(SolutionCompilationState compilationState, CancellationToken cancellationToken)
+            {
+                // If we already computed generator docs, just return this as the frozen tracker.  We wouldn't need to
+                // run generators again anyways.
+                return this;
```

Comment:
When we have this type of compilation tracker, it means the user called GetOpenDocumentInCurrentContextWithChanges() on a text buffer that is a source generated document, and since that function must be fast and cheap, if we can't _immediately_ confirm that the source generated document in the current snapshot matches the text buffer (which may have not updated yet) then we wrap the existing tracker in this thing which says "I don't know about everything else, but I do know this document is there!" So in this case, we don't actually know that generated documents have been computed yet for this project.

I imagine this function should be changed in one of two ways:

1. We convince ourselves it's just not ever going to be called, and throw accordingly.
2. This should probably go to the UnderlyingTracker, freeze _that_, then return a wrapped tracker with the same replacement document, i.e:

```suggestion
                // If we already computed generator docs, just return this as the frozen tracker.  We wouldn't need to
                // run generators again anyways.
                return new GeneratedFileReplacingCompilationTracker(UnderlyingTracker.FreezePartialState(compilationState, cancellationToken), replacementDocumentState);
```

Reply:
sure.